### ISSUE TITLE
Failures in health checking

### ISSUE THREAD

**Issue by Rucknar** [rancher/rancher#2627]
**Server Versions**
```
Component	Version
Rancher	v0.44.0
Cattle	v0.107.0
User Interface	v0.65.0
Rancher Compose	v0.5.1
```
**Host Version**
```
RancherOS 0.4.0-RC-11
```

**Summary**
Services are failing health checks and recycling, after some debugging it appears this could be false test failures from the health checking haproxy node. 

**Background**
Been looking at this with @ibuildthecloud help over the last week, in that time we've fixed a number of issues related to our container builds however the problem remained. I've come up with a repeatable (on our deployment at least) set of steps to replicate the issue. 

**Test Case Summary**
Initiate a build of 100 containers across two hosts (physical in our case) using rancher compose. We've repeated this test with health checking enabled for ports, enabled with a full HTTP1.1 request and disabled. 

**Observed behaviour**
Service builds successfully and containers pass the initial health checks. The services go 'green' in rancher's GUI and the compose prompt returns. After a varying amount of time (generally always sub 60 minutes), a number of the containers all at once will be marked as failed. The docker logs indicates that the rancher load balancer has killed them and rancher takes measures to restore the scaling of the service. I've not found a pattern with how this happens but it occurs ~ every hour. In doing so we can in some scenarios overload docker behind the scenes which causes further issues for us. 

We also see failures on both hosts, usually at the same time as well. We've not been able to identify a pattern but it's never all containers, only ever a subset.

**Docker compose**
```
WebsiteLB:
  ports:
  - 3000:3000
  restart: 'no'
  labels:
    io.rancher.scheduler.global: 'true'
  tty: true
  image: rancher/load-balancer-service
  volumes:
  - /var/log/iam:/var/tmp
  links:
  - 'WebServiceV1:'
  stdin_open: true
WebServiceV1:
  restart: 'no'
  labels:
    io.rancher.container.hostname_override: container_name
  tty: true
  image: n8io/hello-world
  stdin_open: true
  links:
  - 'WebsiteLB:'
```
**Rancher compose** (this one is from when we were testing port checks only).
```
WebsiteLB:
  scale: 2
  load_balancer_config:
    description: Loadbalancer for website testing.
    name: WebsiteLB config
WebServiceV1:
  health_check:
    port: 3000
    interval: 2000
    unhealthy_threshold: 3
    request_line: ''
    healthy_threshold: 2
    response_timeout: 2000
  scale: 100
```

**Docker Events** (from one of two hosts):
```
2015-11-10T11:29:12.000000000Z f4bce0ad12b9c2eee0de9b86fead284be462ad95623d62ab22f6a27156d9024c: (from n8io/hello-world:latest) create
2015-11-10T11:29:12.000000000Z f4bce0ad12b9c2eee0de9b86fead284be462ad95623d62ab22f6a27156d9024c: (from n8io/hello-world:latest) start
2015-11-10T11:29:16.000000000Z 36dba964e5f16a1ce95e380bb71a6a555b24fd201f257e43312f268ed657f45d: (from n8io/hello-world:latest) create
2015-11-10T11:29:17.000000000Z 36dba964e5f16a1ce95e380bb71a6a555b24fd201f257e43312f268ed657f45d: (from n8io/hello-world:latest) start
2015-11-10T11:29:17.000000000Z c85ccaacbd39a50ed1f65f2a708976064b2d31cdea709f3532f81f01edb3ba9d: (from n8io/hello-world:latest) create
2015-11-10T11:29:17.000000000Z c85ccaacbd39a50ed1f65f2a708976064b2d31cdea709f3532f81f01edb3ba9d: (from n8io/hello-world:latest) start
2015-11-10T11:29:17.000000000Z b87d152a3ea4bacfc2a1a2aafb93090b270f08a025d9486f58f9b99946cbfa75: (from n8io/hello-world:latest) create
2015-11-10T11:29:17.000000000Z b87d152a3ea4bacfc2a1a2aafb93090b270f08a025d9486f58f9b99946cbfa75: (from n8io/hello-world:latest) start
2015-11-10T11:29:26.000000000Z 28dd2323f8eb77ea2c5250c42389f85998c21d0bbfefa1ee533bce0d63bd5f37: (from n8io/hello-world:latest) kill
2015-11-10T11:29:29.000000000Z 3cbcba54bcd1506e788e6f145e0f12a512effb61f2f43dc02d7e6ef8bd16bbb2: (from n8io/hello-world:latest) kill
2015-11-10T11:29:36.000000000Z e48783fe904eb79b6c08360d97ff495f14bbee1d60302f6d1a58950463c0b117: (from n8io/hello-world:latest) kill
2015-11-10T11:29:36.000000000Z 28dd2323f8eb77ea2c5250c42389f85998c21d0bbfefa1ee533bce0d63bd5f37: (from n8io/hello-world:latest) kill
2015-11-10T11:29:36.000000000Z 28dd2323f8eb77ea2c5250c42389f85998c21d0bbfefa1ee533bce0d63bd5f37: (from n8io/hello-world:latest) die
2015-11-10T11:29:36.000000000Z 28dd2323f8eb77ea2c5250c42389f85998c21d0bbfefa1ee533bce0d63bd5f37: (from n8io/hello-world:latest) stop
2015-11-10T11:29:39.000000000Z 3cbcba54bcd1506e788e6f145e0f12a512effb61f2f43dc02d7e6ef8bd16bbb2: (from n8io/hello-world:latest) kill
2015-11-10T11:29:40.000000000Z 3cbcba54bcd1506e788e6f145e0f12a512effb61f2f43dc02d7e6ef8bd16bbb2: (from n8io/hello-world:latest) die
2015-11-10T11:29:40.000000000Z 3cbcba54bcd1506e788e6f145e0f12a512effb61f2f43dc02d7e6ef8bd16bbb2: (from n8io/hello-world:latest) stop
2015-11-10T11:29:46.000000000Z e48783fe904eb79b6c08360d97ff495f14bbee1d60302f6d1a58950463c0b117: (from n8io/hello-world:latest) kill
2015-11-10T11:29:46.000000000Z e48783fe904eb79b6c08360d97ff495f14bbee1d60302f6d1a58950463c0b117: (from n8io/hello-world:latest) die
2015-11-10T11:29:46.000000000Z e48783fe904eb79b6c08360d97ff495f14bbee1d60302f6d1a58950463c0b117: (from n8io/hello-world:latest) stop
```

**Haproxy Logs**:
```
time="2015-11-10T11:28:58Z" level="info" msg="5687d5c9-e725-49d9-a277-2a130423f5a8_0=UP"
time="2015-11-10T11:28:59Z" level="info" msg="cf37561c-fa93-4021-bf5c-ffac2409340a_0=UP"
time="2015-11-10T11:29:00Z" level="info" msg="31a19ee5-9b33-40e2-9278-51834a1c242b_0=UP"
time="2015-11-10T11:29:00Z" level="info" msg="ee7c5e02-ad95-41c8-b11c-e1e13dcf256b_0=UP"
time="2015-11-10T11:29:01Z" level="info" msg="d6bad138-5eed-428d-9256-c6c9fe172a83_0=UP"
time="2015-11-10T11:29:02Z" level="info" msg="f166e2f8-aee6-41f0-9243-45626dee67f0_0=UP"
time="2015-11-10T11:29:03Z" level="info" msg="571ad227-4df1-43ff-babd-76f0611e34d1_1=UP"
time="2015-11-10T11:29:03Z" level="info" msg="31553c25-a54b-417a-a820-5674e8cbacd5_1=UP"
time="2015-11-10T11:29:04Z" level="info" msg="f27d4e6e-5297-4b20-9bda-4256d833177b_1=UP"
time="2015-11-10T11:29:06Z" level="info" msg="e7510e7a-874d-4c64-a69b-6045603fd144_0=DOWN"
time="2015-11-10T11:29:06Z" level="info" msg="f981ddea-2e6b-49ad-8637-0113c774d4a6_0=DOWN"
time="2015-11-10T11:29:06Z" level="info" msg="f972ac4a-ff20-4568-a1de-d58a89dfb45b_0=DOWN"
time="2015-11-10T11:29:06Z" level="info" msg="Monitoring 53 backends"
time="2015-11-10T11:29:07Z" level="info" msg="a21e8980-9b8f-4c5d-821f-bf34b2db8f3d_0=DOWN"
time="2015-11-10T11:29:12Z" level="info" msg="Monitoring 55 backends"
time="2015-11-10T11:29:17Z" level="info" msg="23c224cb-f77d-42a9-acf8-46750154511c_0=DOWN"
time="2015-11-10T11:29:17Z" level="info" msg="3695b608-6224-4b16-bebd-5f0a4c011f4c_0=DOWN"
time="2015-11-10T11:29:19Z" level="info" msg="e7510e7a-874d-4c64-a69b-6045603fd144_0=DOWN 1/2"
time="2015-11-10T11:29:19Z" level="info" msg="f972ac4a-ff20-4568-a1de-d58a89dfb45b_0=DOWN 1/2"
time="2015-11-10T11:29:20Z" level="info" msg="3695b608-6224-4b16-bebd-5f0a4c011f4c_0=DOWN 1/2"
time="2015-11-10T11:29:22Z" level="info" msg="e7510e7a-874d-4c64-a69b-6045603fd144_0=INIT"
time="2015-11-10T11:29:22Z" level="info" msg="f981ddea-2e6b-49ad-8637-0113c774d4a6_0=INIT"
time="2015-11-10T11:29:22Z" level="info" msg="f972ac4a-ff20-4568-a1de-d58a89dfb45b_0=INIT"
time="2015-11-10T11:29:23Z" level="info" msg="a21e8980-9b8f-4c5d-821f-bf34b2db8f3d_0=INIT"
time="2015-11-10T11:29:23Z" level="info" msg="23c224cb-f77d-42a9-acf8-46750154511c_0=INIT"
time="2015-11-10T11:29:24Z" level="info" msg="3695b608-6224-4b16-bebd-5f0a4c011f4c_0=INIT"
time="2015-11-10T11:29:26Z" level="info" msg="e7510e7a-874d-4c64-a69b-6045603fd144_0=UP"
time="2015-11-10T11:29:26Z" level="info" msg="f981ddea-2e6b-49ad-8637-0113c774d4a6_0=UP"
time="2015-11-10T11:29:26Z" level="info" msg="f972ac4a-ff20-4568-a1de-d58a89dfb45b_0=UP"
time="2015-11-10T11:29:26Z" level="info" msg="a21e8980-9b8f-4c5d-821f-bf34b2db8f3d_0=UP"
time="2015-11-10T11:29:26Z" level="info" msg="23c224cb-f77d-42a9-acf8-46750154511c_0=UP"
time="2015-11-10T11:29:26Z" level="info" msg="3695b608-6224-4b16-bebd-5f0a4c011f4c_0=UP"
```
**Rancher Server logs**
Not posted as it's just lots of logs about DB records changing

**Cattle logs**
```
2015-11-10 11:34:31,828 INFO  [:] [] [] [] [cutorService-18] [i.c.p.a.s.s.impl.AgentServiceImpl   ] Timeout waiting for response to [config.update] id [7e8da12a-c3af-427b-a2b9-56c08d55501a]
2015-11-10 11:34:31,828 INFO  [:] [] [] [] [cutorService-18] [i.c.p.a.s.s.impl.AgentServiceImpl   ] Timeout waiting for response to [config.update] id [5797951f-1671-4edb-aadb-99a027d63494]
2015-11-10 11:34:31,828 INFO  [:] [] [] [] [cutorService-18] [i.c.p.a.s.s.impl.AgentServiceImpl   ] Timeout waiting for response to [delegate.request] id [9e148053-da6c-495a-be20-e3c37bb2a92e]
2015-11-10 11:34:31,828 INFO  [:] [] [] [] [cutorService-18] [i.c.p.a.s.s.impl.AgentServiceImpl   ] Timeout waiting for response to [ping] id [00ba6e77-6288-4b73-81f4-fd9d9c0d9f8b]
2015-11-10 11:34:31,836 INFO  [:] [] [] [] [ecutorService-1] [.p.c.v.i.ConfigItemStatusManagerImpl] Timeout updating item(s) [healthcheck, ipsec-hosts, hosts, iptables] on [agent:56]
2015-11-10 11:34:31,836 INFO  [:] [] [] [] [ecutorService-1] [.p.c.v.i.ConfigItemStatusManagerImpl] Timeout updating item(s) [healthcheck, ipsec-hosts, hosts, iptables] on [agent:56]
2015-11-10 11:34:31,843 INFO  [:] [] [] [] [ecutorService-1] [.p.c.v.i.ConfigItemStatusManagerImpl] Timeout updating item(s) [host-routes] on [agent:53]
2015-11-10 11:34:31,843 INFO  [:] [] [] [] [ecutorService-1] [.p.c.v.i.ConfigItemStatusManagerImpl] Timeout updating item(s) [host-routes] on [agent:53]
```

**Variations**
- We used this with HTTP1.1 checking initially as that matches our current stack.
- This happens with our own image, tutum/hello-world & this node based test container.

**Considerations**
- We've upgraded rancher a number of times, DB mapped in from the host.
- Our cattle logs show a concerning amount of timeout errors
- We've tested connectivity between Rancher Server and the Host / Healthcheck agent and the Container.We've not managed to find any packet loss or connectivity issues. 
- Rancher server lives on a VM running on a separate host.

**Comment by ibuildthecloud** [comment]
@alena1108 PTAL

**Comment by alena1108** [comment]
@Rucknar thank you for detailed analysis. Could you please elaborate the following:

>> We also see failures on both hosts, usually at the same time as well. We've not been able to identify a pattern but it's never all containers, only ever a subset.

What kind of failures have you seen besides those failed healtchecks? Are these failures in some way related to a failed network? 

>> We've tested connectivity between Rancher Server and the Host / Healthcheck agent and the Container.We've not managed to find any packet loss or connectivity issues.

Have you tested it around the time when the healthcheck failures start occurring? If not, is is possible to do so around the time?

>> Our cattle logs show a concerning amount of timeout errors

Do these errors happen around the same time failed health checks happen?


We have to get down to the root cause of the failures on your system as it looks concerning. But as a possible workaround for your system, you can try to increase values for:

unhealthyThreshold
responseTimeout

**Comment by Rucknar** [comment]
@alena1108 Sure thing,
>What kind of failures have you seen besides those failed healtchecks? Are these failures in some way related to a failed network?

In this scenario i'm seeing containers marked as down across both of our hosts at the same or a very close time. The hosts themselves reside on an existing 10GB network, i'm not aware or able to reproduce any network issues. I feel a network issue with the switch to be unlikely at this stage.

>Have you tested it around the time when the healthcheck failures start occurring? If not, is is possible to do so around the time?

I've run pings and netcat's to the ports of a few containers, due to it's random nature i've not managed to do this to a container that's restarted. 

> Do these errors happen around the same time failed health checks happen?

These are constant, we've stood our application stack up this afternoon without healthchecks to enable our functional testing to proceed, all is stable and not much is going on but these timeouts are streaming past out of the cattle logs.

To elaborate further, below is some more information i've captured this eveing. The graphs show the previous 15 minute period, i can show the previous 6 hours of data if required to demonstrate it's 'resting' performance. I'll post this info up now and then some more information in the next hour or two once the problem has manifested itsself. 

20:37 - Initiated a 'up' with the above config
20:45 - compose returned and service stated as active. 

<img width="838" alt="screen shot 2015-11-10 at 20 48 49" src="https://cloud.githubusercontent.com/assets/5539660/11075490/4a4a9b3c-87ed-11e5-8933-47f0b72ee250.png">
<img width="826" alt="screen shot 2015-11-10 at 20 48 45" src="https://cloud.githubusercontent.com/assets/5539660/11075491/4a4b6daa-87ed-11e5-9bd9-36e7c4e2907b.png">
<img width="841" alt="screen shot 2015-11-10 at 20 48 38" src="https://cloud.githubusercontent.com/assets/5539660/11075492/4a4c2416-87ed-11e5-9d83-6ec2c505c983.png">
<img width="840" alt="screen shot 2015-11-10 at 20 48 33" src="https://cloud.githubusercontent.com/assets/5539660/11075493/4a4c29b6-87ed-11e5-8fdc-b52ab4be4818.png">
<img width="844" alt="screen shot 2015-11-10 at 20 48 27" src="https://cloud.githubusercontent.com/assets/5539660/11075495/4a4d6fd8-87ed-11e5-9a22-c302b4e9fa6f.png">
<img width="833" alt="screen shot 2015-11-10 at 20 48 16" src="https://cloud.githubusercontent.com/assets/5539660/11075494/4a4cc164-87ed-11e5-9a3c-67b768e4c602.png">
<img width="829" alt="screen shot 2015-11-10 at 20 48 10" src="https://cloud.githubusercontent.com/assets/5539660/11075496/4a5e9f56-87ed-11e5-9faf-2be40e483d40.png">
<img width="830" alt="screen shot 2015-11-10 at 20 48 02" src="https://cloud.githubusercontent.com/assets/5539660/11075499/4a634704-87ed-11e5-8dce-929e33dff333.png">

**Comment by Rucknar** [comment]
^ To add to the above, i'm wondering if the timeouts are related to a number of stuck process's i can see in the process tab within rancher. These are simple normal tasks that for whatever reason have persisted restarts of the container hosts and of the rancher server its-self. Apologies i didn't mention this before, it's been a long week already!

<img width="1669" alt="screen shot 2015-11-10 at 21 08 52" src="https://cloud.githubusercontent.com/assets/5539660/11075828/476f671a-87ef-11e5-86bc-1a16051b4eb7.png">

**Comment by Rucknar** [comment]
Just had the failure conditions occur following the start of that stack. 

**Host 1 docker events**
```
2015-11-10T21:34:18.000000000Z 16105c6c3c634fa0af176546fc52c7e449721a0ee42f2f228bc36c0b5714ea78: (from n8io/hello-world:latest) create
2015-11-10T21:34:19.000000000Z 16105c6c3c634fa0af176546fc52c7e449721a0ee42f2f228bc36c0b5714ea78: (from n8io/hello-world:latest) start
2015-11-10T21:34:22.000000000Z a19666a12e66746107bb9e574fa341d9e2d1748f6707e7e5dba7baba84d56171: (from n8io/hello-world:latest) create
2015-11-10T21:34:23.000000000Z a19666a12e66746107bb9e574fa341d9e2d1748f6707e7e5dba7baba84d56171: (from n8io/hello-world:latest) start
2015-11-10T21:34:38.000000000Z 46b9e1331ef3daa1fa3469b97f4aaaf860aed5882ac4f81ef3af814682efed9b: (from n8io/hello-world:latest) kill
2015-11-10T21:34:40.000000000Z 48044cbfb8ed9218940aaa003adb15eb620d7d686818805c31cb1e47f79b94a0: (from n8io/hello-world:latest) kill
2015-11-10T21:34:48.000000000Z 46b9e1331ef3daa1fa3469b97f4aaaf860aed5882ac4f81ef3af814682efed9b: (from n8io/hello-world:latest) kill
2015-11-10T21:34:50.000000000Z 48044cbfb8ed9218940aaa003adb15eb620d7d686818805c31cb1e47f79b94a0: (from n8io/hello-world:latest) kill
2015-11-10T21:34:51.000000000Z 46b9e1331ef3daa1fa3469b97f4aaaf860aed5882ac4f81ef3af814682efed9b: (from n8io/hello-world:latest) die
2015-11-10T21:34:53.000000000Z 48044cbfb8ed9218940aaa003adb15eb620d7d686818805c31cb1e47f79b94a0: (from n8io/hello-world:latest) die
2015-11-10T21:35:09.000000000Z 46b9e1331ef3daa1fa3469b97f4aaaf860aed5882ac4f81ef3af814682efed9b: (from n8io/hello-world:latest) stop
2015-11-10T21:35:09.000000000Z 48044cbfb8ed9218940aaa003adb15eb620d7d686818805c31cb1e47f79b94a0: (from n8io/hello-world:latest) stop
```
** Host 2 Docker stats**
```
2015-11-10T21:34:13.000000000Z 37d1dcffa223d002dd32ab4265d83d44fd461301ddab441471dd0a708ce1cb92: (from n8io/hello-world:latest) create
2015-11-10T21:34:14.000000000Z 37d1dcffa223d002dd32ab4265d83d44fd461301ddab441471dd0a708ce1cb92: (from n8io/hello-world:latest) start
2015-11-10T21:34:15.000000000Z 7937f02a57b214d8070cd9450ba13ac564707575abd6045b7bde6a70c5da6ad1: (from n8io/hello-world:latest) create
2015-11-10T21:34:15.000000000Z 7937f02a57b214d8070cd9450ba13ac564707575abd6045b7bde6a70c5da6ad1: (from n8io/hello-world:latest) start
2015-11-10T21:34:29.000000000Z 6c25d63eaa473ce19e4a6f7133189b6e1447a4243bd0eff9014ee16da8538954: (from n8io/hello-world:latest) kill
2015-11-10T21:34:33.000000000Z a7b87ea2c3c7408957de5a12a173a078b88d3f591967f43c9236412622aedb13: (from n8io/hello-world:latest) kill
2015-11-10T21:34:33.000000000Z c865a339f59125f7cf7290eb322189655659123bc2891ac0f333af16b8adc996: (from n8io/hello-world:latest) kill
2015-11-10T21:34:35.000000000Z fa6c9446a47b9b67a6750a654cb711d338ccfb107837bbe3702dc22a7accbe9f: (from n8io/hello-world:latest) kill
2015-11-10T21:34:36.000000000Z 625961597b609f031682c98e901e97988ff0ef4749f0d70dd21f7e56a0f5462b: (from n8io/hello-world:latest) create
2015-11-10T21:34:36.000000000Z 169b89d310cf119e9b04326b24100adbb2f7f7dd6fa9dc84f56e47d96ad66cdf: (from n8io/hello-world:latest) create
2015-11-10T21:34:36.000000000Z 625961597b609f031682c98e901e97988ff0ef4749f0d70dd21f7e56a0f5462b: (from n8io/hello-world:latest) start
2015-11-10T21:34:36.000000000Z 169b89d310cf119e9b04326b24100adbb2f7f7dd6fa9dc84f56e47d96ad66cdf: (from n8io/hello-world:latest) start
2015-11-10T21:34:37.000000000Z 0f5dbe9fae17b3889b0669ce3d9bb118a3e18d7a505bc8c4e7fa35582194dd83: (from n8io/hello-world:latest) create
2015-11-10T21:34:37.000000000Z 9b24cec3fd37b57804504537e85453827e0e94ba6f4a7ee5a06297e4ef1f6435: (from n8io/hello-world:latest) kill
2015-11-10T21:34:37.000000000Z 02a0a10ffe6e1e8d8109433c96d75cff8dc41cbb090b4aab491ef05e4b63c2a8: (from n8io/hello-world:latest) create
2015-11-10T21:34:37.000000000Z 0f5dbe9fae17b3889b0669ce3d9bb118a3e18d7a505bc8c4e7fa35582194dd83: (from n8io/hello-world:latest) start
2015-11-10T21:34:37.000000000Z 02a0a10ffe6e1e8d8109433c96d75cff8dc41cbb090b4aab491ef05e4b63c2a8: (from n8io/hello-world:latest) start
2015-11-10T21:34:39.000000000Z 6c25d63eaa473ce19e4a6f7133189b6e1447a4243bd0eff9014ee16da8538954: (from n8io/hello-world:latest) kill
2015-11-10T21:34:39.000000000Z 6c25d63eaa473ce19e4a6f7133189b6e1447a4243bd0eff9014ee16da8538954: (from n8io/hello-world:latest) die
2015-11-10T21:34:39.000000000Z 6c25d63eaa473ce19e4a6f7133189b6e1447a4243bd0eff9014ee16da8538954: (from n8io/hello-world:latest) stop
2015-11-10T21:34:42.000000000Z 625961597b609f031682c98e901e97988ff0ef4749f0d70dd21f7e56a0f5462b: (from n8io/hello-world:latest) kill
2015-11-10T21:34:43.000000000Z a7b87ea2c3c7408957de5a12a173a078b88d3f591967f43c9236412622aedb13: (from n8io/hello-world:latest) kill
2015-11-10T21:34:43.000000000Z a7b87ea2c3c7408957de5a12a173a078b88d3f591967f43c9236412622aedb13: (from n8io/hello-world:latest) die
2015-11-10T21:34:43.000000000Z a7b87ea2c3c7408957de5a12a173a078b88d3f591967f43c9236412622aedb13: (from n8io/hello-world:latest) stop
2015-11-10T21:34:43.000000000Z c865a339f59125f7cf7290eb322189655659123bc2891ac0f333af16b8adc996: (from n8io/hello-world:latest) kill
2015-11-10T21:34:44.000000000Z c865a339f59125f7cf7290eb322189655659123bc2891ac0f333af16b8adc996: (from n8io/hello-world:latest) die
2015-11-10T21:34:44.000000000Z c865a339f59125f7cf7290eb322189655659123bc2891ac0f333af16b8adc996: (from n8io/hello-world:latest) stop
2015-11-10T21:34:45.000000000Z fa6c9446a47b9b67a6750a654cb711d338ccfb107837bbe3702dc22a7accbe9f: (from n8io/hello-world:latest) kill
2015-11-10T21:34:45.000000000Z fa6c9446a47b9b67a6750a654cb711d338ccfb107837bbe3702dc22a7accbe9f: (from n8io/hello-world:latest) die
2015-11-10T21:34:45.000000000Z fa6c9446a47b9b67a6750a654cb711d338ccfb107837bbe3702dc22a7accbe9f: (from n8io/hello-world:latest) stop
2015-11-10T21:34:47.000000000Z 9b24cec3fd37b57804504537e85453827e0e94ba6f4a7ee5a06297e4ef1f6435: (from n8io/hello-world:latest) kill
2015-11-10T21:34:47.000000000Z 9b24cec3fd37b57804504537e85453827e0e94ba6f4a7ee5a06297e4ef1f6435: (from n8io/hello-world:latest) die
2015-11-10T21:34:47.000000000Z 9b24cec3fd37b57804504537e85453827e0e94ba6f4a7ee5a06297e4ef1f6435: (from n8io/hello-world:latest) stop
2015-11-10T21:34:52.000000000Z 625961597b609f031682c98e901e97988ff0ef4749f0d70dd21f7e56a0f5462b: (from n8io/hello-world:latest) kill
2015-11-10T21:34:52.000000000Z 625961597b609f031682c98e901e97988ff0ef4749f0d70dd21f7e56a0f5462b: (from n8io/hello-world:latest) die
2015-11-10T21:34:52.000000000Z 625961597b609f031682c98e901e97988ff0ef4749f0d70dd21f7e56a0f5462b: (from n8io/hello-world:latest) stop
```
Some semi-useful graphs from around the time of the failure, i'd happily email you over the logs for cattle if you wish. 

<img width="841" alt="screen shot 2015-11-10 at 21 39 26" src="https://cloud.githubusercontent.com/assets/5539660/11076688/01a86c04-87f4-11e5-83e0-9aae8fbfbdfe.png">
<img width="840" alt="screen shot 2015-11-10 at 21 39 20" src="https://cloud.githubusercontent.com/assets/5539660/11076685/01a3bf56-87f4-11e5-95db-01baf293bead.png">
<img width="831" alt="screen shot 2015-11-10 at 21 39 08" src="https://cloud.githubusercontent.com/assets/5539660/11076687/01a771aa-87f4-11e5-887e-5b858c143a49.png">
<img width="822" alt="screen shot 2015-11-10 at 21 39 02" src="https://cloud.githubusercontent.com/assets/5539660/11076686/01a767c8-87f4-11e5-9719-9b9cd4055f6a.png">

**Comment by alena1108** [comment]
>> In this scenario i'm seeing containers marked as down across both of our hosts at the same or a very close time. 

Are those containers the ones that got killed by the health checks monitor? Or those are the containers that were marked as down besides the containers that had health checks failed. Wonder if only health check related "kills" happen every ~60 mins, or there is some other bad activity happening around the same time.

>> I've run pings and netcat's to the ports of a few containers, due to it's random nature i've not managed to do this to a container that's restarted.

Is it possible to write a small script performing this check, and run it overtime? That would test the generic network connectivity. It would also be awesome if we could test the ipsec connectivity as well. For ipsec, we would need to run ping test inside network-agent on each host. We should consistently ping  the 10.42.x.x ip address of the network agent on another host and see if it ever fails. Want to eliminate possibility of ipsec network glitch that could have caused the failed health check. In this case failed health check would be a symptom of a bigger issue which you wouldn't be able to catch otherwise as ipsec failure doesn't result in container start failure, but connectivity between containers gets broken.


>> To add to the above, i'm wondering if the timeouts are related to a number of stuck process's i can see in the process tab within rancher.

Its actually the other way around. Config scripts update timeouts caused processes being stuck. 

Thank you for the graphs. Most likely the spikes happen around the time when all containers with the failed health check are being restarted.

**Comment by JamesBarwell** [comment]
Hi, I'm working with @Rucknar on this. We have performed tests to check both our network and the IPsec network. On the basis of these tests we do not think there are problems with either network.

## Test script

We used a script to exec ping commands on the appropriate containers, which is available here: https://gist.github.com/JamesBarwell/a3758a292cf764d414cd

## Test set up

We took readings both against our RancherOS hosts, and to ping the counterpart agent as requested.

Our network:
Rancher server -> Host A
Rancher server -> Host B
Rancher agent (Host A) -> Host B
Rancher agent (Host B) -> Host A

IPSec tunnel:
Rancher agent (Host A) -> Rancher agent (Host B)
Rancher agent (Host B) -> Rancher agent (Host A)

We took readings while running a stack with health-checks (and during the run, these health-checks failed as @Rucknar described). We also took some baseline readings where no health-checked stack was running. We ran all of these for one hour - i.e. 18,000 packets at a 200ms interval, for each of the above configurations.

## Results

We saw no significant difference in all of the tests, whether running the stack or not. We saw ping times fairly steady in the 0-1ms range, and only very occasionally creeping up to 10-25ms (of 18,000 pings in one run, only around 5 were in the double-digit range). We only saw a single packet go missing from Rancher agent (Host A) to Rancher agent (Host B).

```
# Rancher server to each host
[server] --- <hostA IP> ping statistics ---
[server] 18000 packets transmitted, 18000 received, 0% packet loss, time 3604188ms
[server] rtt min/avg/max/mdev = 0.205/0.574/22.579/0.416 ms

[server] --- <hostB IP> ping statistics ---
[server] 18000 packets transmitted, 18000 received, 0% packet loss, time 3603942ms
[server] rtt min/avg/max/mdev = 0.206/0.570/5.571/0.345 ms

# Rancher agent to each host
[hostA] --- <hostB IP> ping statistics ---
[hostA] 18000 packets transmitted, 18000 received, 0% packet loss, time 3603563ms
[hostA] rtt min/avg/max/mdev = 0.059/0.315/17.436/0.591 ms

[hostB] --- <hostA IP> ping statistics ---
[hostB] 18000 packets transmitted, 18000 received, 0% packet loss, time 3603366ms
[hostB] rtt min/avg/max/mdev = 0.058/0.330/17.494/0.595 ms

# Rancher agent to the other rancher agent
[hostA] --- <Rancher agent Host B IP: 10.42.160.94> ping statistics ---
[hostA] 18000 packets transmitted, 17999 received, 0% packet loss, time 3603416ms
[hostA] rtt min/avg/max/mdev = 0.094/0.356/25.776/0.687 ms

[hostB] --- <Rancher agent Host A IP: 10.42.113.67> ping statistics ---
[hostB] 18000 packets transmitted, 18000 received, 0% packet loss, time 3602918ms
[hostB] rtt min/avg/max/mdev = 0.099/0.394/21.452/0.717 ms
```

**Comment by Rucknar** [comment]
During that test i captured the cattle error log and debug log, i've provided an excerpt of these logs during the period whilst containers were restarting, ignore if it's not useful of course.

**Error**
```
2015-11-11 13:01:38,574 ERROR [b3e772cd-f6db-49e1-a7a0-f87f4a3b73cb:1006881] [service:1975] [service.remove] [] [cutorService-54] [c.p.e.p.i.DefaultProcessInstanceImpl] Unknown exception io.cattle.platform.engine.process.ProcessInstanceException: io.cattle.platform.engine.process.impl.ProcessExecutionExitException: PROCESS_ALREADY_IN_PROGRESS
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.execute(DefaultProcessInstanceImpl.java:115) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.manager.impl.DefaultProcessManager.scheduleProcessInstance(DefaultProcessManager.java:60) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.object.process.impl.DefaultObjectProcessManager.scheduleProcessInstance(DefaultObjectProcessManager.java:47) ~[cattle-framework-object-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.object.process.impl.DefaultObjectProcessManager.scheduleStandardProcess(DefaultObjectProcessManager.java:40) ~[cattle-framework-object-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.object.process.impl.DefaultObjectProcessManager.scheduleStandardProcess(DefaultObjectProcessManager.java:34) ~[cattle-framework-object-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.object.process.impl.DefaultObjectProcessManager$2.run(DefaultObjectProcessManager.java:126) ~[cattle-framework-object-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.deferred.util.DeferredUtils$3.call(DeferredUtils.java:43) ~[cattle-framework-deferred-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.deferred.util.DeferredUtils.nest(DeferredUtils.java:53) ~[cattle-framework-deferred-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.deferred.util.DeferredUtils.nest(DeferredUtils.java:40) ~[cattle-framework-deferred-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.object.process.impl.DefaultObjectProcessManager.scheduleStandardProcessAsync(DefaultObjectProcessManager.java:123) ~[cattle-framework-object-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.servicediscovery.deployment.impl.DefaultDeploymentUnitInstance.removeUnitInstance(DefaultDeploymentUnitInstance.java:65) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.servicediscovery.deployment.DeploymentUnitInstance.remove(DeploymentUnitInstance.java:28) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.servicediscovery.deployment.impl.DeploymentUnit.remove(DeploymentUnit.java:151) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl$3.deleteServiceInstances(DeploymentManagerImpl.java:263) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl$3.doWithLockNoResult(DeploymentManagerImpl.java:250) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl.remove(DeploymentManagerImpl.java:246) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.servicediscovery.process.ServiceRemove.handle(ServiceRemove.java:43) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runHandler(DefaultProcessInstanceImpl.java:456) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$4.execute(DefaultProcessInstanceImpl.java:396) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$4.execute(DefaultProcessInstanceImpl.java:390) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.idempotent.Idempotent.execute(Idempotent.java:42) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runHandlers(DefaultProcessInstanceImpl.java:390) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runLogic(DefaultProcessInstanceImpl.java:520) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runWithProcessLock(DefaultProcessInstanceImpl.java:323) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$2.doWithLockNoResult(DefaultProcessInstanceImpl.java:259) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.acquireLockAndRun(DefaultProcessInstanceImpl.java:256) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runDelegateLoop(DefaultProcessInstanceImpl.java:188) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.executeWithProcessInstanceLock(DefaultProcessInstanceImpl.java:161) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:110) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:107) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.execute(DefaultProcessInstanceImpl.java:107) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.eventing.impl.ProcessEventListenerImpl.processExecute(ProcessEventListenerImpl.java:66) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at sun.reflect.GeneratedMethodAccessor480.invoke(Unknown Source) ~[na:na]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_85]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_85]
        at io.cattle.platform.eventing.annotation.MethodInvokingListener$1.doWithLockNoResult(MethodInvokingListener.java:76) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.eventing.annotation.MethodInvokingListener.onEvent(MethodInvokingListener.java:72) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.eventing.impl.AbstractThreadPoolingEventService$2.doRun(AbstractThreadPoolingEventService.java:135) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.NoExceptionRunnable.runInContext(NoExceptionRunnable.java:13) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_85]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_85]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_85]
Caused by: io.cattle.platform.engine.process.impl.ProcessExecutionExitException: PROCESS_ALREADY_IN_PROGRESS
        ... 69 common frames omitted
```

**Debug Log attached.**
[2672debug.txt](https://github.com/rancher/rancher/files/31883/2672debug.txt)

**Comment by alena1108** [comment]
@JamesBarwell @Rucknar thank you for running all these tests, its very helpful. I'm going to work with Rancher QA team to see if we can reproduce the issue in house. 

I also know that your logs are really huge, but could you upload the log snippet taken around all these failures occur (with half an hour shift before and after the failures had happened), and send me a link to it? My email is alena at rancher.com.

**Comment by alena1108** [comment]
@JamesBarwell @Rucknar was debugging very similar issue, with the smaller scale though (https://github.com/rancher/rancher/issues/2637, check the last comment)

For any of the instances that got purged (and later on replaced by the new instance), could you run this query:

select id, reported_health, host_id, created from service_event where instance_id=<instance_id> order by id;

And see if you observe this kind of pattern - INIT followed up by UP and then INIT again

mysql> select reported_health from service_event where instance_id=3;
+-----------------+
| reported_health |
+-----------------+
| DOWN            |
| DOWN 1/2        |
| INIT            |
| UP              |
| INIT            |
| UP              |
+-----------------+

Let me know if thats what you see happening on your setup.

**Comment by Rucknar** [comment]
Good shout... We also do notice a number of them get recycled before coming up clean at first.
Below is the output for one such node, i'm going to re-run this in the next hour when we typically see them fail and i'll update the ticket.

```
mysql> select id, reported_health, host_id, created from service_event where instance_id=35393 order by id;
+-------+-----------------+---------+---------------------+
| id    | reported_health | host_id | created             |
+-------+-----------------+---------+---------------------+
| 72683 | DOWN            |       6 | 2015-11-12 08:28:29 |
| 72696 | DOWN 1/2        |       6 | 2015-11-12 08:28:40 |
| 72698 | INIT            |       6 | 2015-11-12 08:28:42 |
| 72699 | UP              |       6 | 2015-11-12 08:28:44 |
| 72701 | INIT            |       6 | 2015-11-12 08:28:53 |
| 72709 | UP              |       6 | 2015-11-12 08:28:56 |
+-------+-----------------+---------+---------------------+
6 rows in set (0.00 sec)
```

**Comment by Rucknar** [comment]
And for a container that failed after 50 minutes (ish) of running
```
mysql> select id, reported_health, host_id, created from service_event where instance_id=35411 order by id;
+-------+-----------------+---------+---------------------+
| id    | reported_health | host_id | created             |
+-------+-----------------+---------+---------------------+
| 72838 | DOWN            |       6 | 2015-11-12 08:30:08 |
| 72851 | DOWN 1/2        |       6 | 2015-11-12 08:30:15 |
| 72855 | INIT            |       6 | 2015-11-12 08:30:17 |
| 72857 | UP              |       6 | 2015-11-12 08:30:19 |
| 72918 | INIT            |       6 | 2015-11-12 09:20:58 |
| 72971 | UP              |       6 | 2015-11-12 09:21:27 |
+-------+-----------------+---------+---------------------+
6 rows in set (0.00 sec)
```

Sounds like this is very likely to be the bug.

**Comment by alena1108** [comment]
@Rucknar yup, loos exactly the same. UP -> INIT should never result in container being recycled. If you look at the last example, you see that the first UP was at  08:30:19, and then ~one hour later second INIT followed (09:20:58). That was the trigger point for container being killed.

**Comment by alena1108** [comment]
PR ref: https://github.com/rancher/cattle/pull/1045

**Comment by alena1108** [comment]
@sangeethah I know we have a setup replicating @Rucknar @JamesBarwell environment, and that the issue was reproduced yesterday night - thanks for working on it. Lets run the same kind of test against the master build to see if it still happens - the fix was merged to master yesterday.

**Comment by Rucknar** [comment]
We're happy to test a build too if you wish?

**Comment by sangeethah** [comment]
@alena1108 @Rucknar , I was able to run the similar kind of test with a build that has fox where I have the following 2 services running for the last 4 hours:

1. service  of scale 100 with http health check enabled 
2. service  of scale 100 with tcp health check enabled 

I have not seen instances being marked "unhealthy" and purged in the last few hours ( There were few instances that were purged when service was building the scale of 100 . But none after the service was fully scaled.) 

Will leave the setup tonight and report back my findings.

**Comment by sangeethah** [comment]
After letting the server up and running for the last 20 hours , there have been no additional  instances being marked "unhealthy" and purged. 
Will close this issue.

@Rucknar , the fix for this issue will be available in the next RC candidate that will be made available. Please reopen this issue if you still see it with RC build that has this fix.

**Comment by Rucknar** [comment]
Thanks @ibuildthecloud @alena1108 @sangeethah This was a really important for us, were now unblocked and can continue on our path to live, appreciate the hard work put in to get it sorted so fast.

**Comment by alena1108** [comment]
@Rucknar thank you and @JamesBarwell on raising the issue and great help debugging it.
